src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/ReadOnlyListExtensions.cs

```diff
@@ -880,30 +880,20 @@ public static T SingleOrDefault<T>(this IReadOnlyList<T> list, Func<T, bool> pre
 
     public static Enumerable<T> AsEnumerable<T>(this IReadOnlyList<T> list) => new(list);
 
-    public readonly struct Enumerable<T>(IReadOnlyList<T> list) : IEnumerable<T>
+    public readonly ref struct Enumerable<T>(IReadOnlyList<T> list)
```

Comment:
What's the benefit of changing these to `ref struct` types?

Reply:
It came out of @jjonescz's comment on a different PR [here](https://github.com/dotnet/razor/pull/10602#discussion_r1677429927). Essentially, there's no reason to ever want to box these types. So, it's just a little extra protection. I don't think there's really much value in this change, but it was simple enough. It's just a little extra protection. After all, if the user wants to pass an `IReadOnlyList<T>` to an `IEnumerable<T>`, they can just pass it.